core/client/app/components/gh-ed-editor.js

```diff
@@ -50,6 +52,75 @@ export default Ember.TextArea.extend(EditorAPI, EditorShortcuts, EditorScroll, {
     },
 
     /**
+     * Binds the paste and drop events on the editor
+     */
+    attachFileHandler: function () {
+        var self = this,
+            fileUpload = this.$().fileupload(),
+            dropSettings = {
+                progressbar: true,
+                editor: false
+            },
+            mimeTypes = {
+                'image/jpeg': 'jpg',
+                'image/png': 'png',
+                'image/gif': 'gif',
+                'image/svg+xml': 'svg'
+            },
+            latestUpload;
+
+        fileUpload.fileupload('option', {
+            url: ghostPaths().apiRoot + '/uploads/',
+            pasteZone: this.$(),
+            dropZone: this.$(),
+            paramName: 'uploadimage',
+            add: function (e, data) {
+                var selection = self.getSelection();
+                self.replaceSelection('![uploading...]()', selection.start, selection.end, 'collapseToEnd');
+                latestUpload = new UploadUi($('.js-drop-zone:contains("uploading...")'), dropSettings);
+                latestUpload.initProgress(data);
+            },
+            submit: function (e, data) {
+                var ext = '.jpg',
+                    file = data.files[0];
+
+                if (file.type && mimeTypes[file.type]) {
+                    ext =  mimeTypes[file.type];
+                }
+
+                data.formData = {
+                    filename: 'paste.' + ext
+                };
+            },
+            paste: function (e, data) {
+                self.$().fileupload('add', {files: data.files});
+                e.preventDefault();
+            },
+            fail: function (e, data) {
+                latestUpload.setError(data);
+            },
+            progressall: function (e, data) {
+                latestUpload.setProgress(data);
+            },
+            done: function (e, data) {
+                var filename = 'image',
+                    result = data.result;
+
+                if (result) {
+                    filename = result.substring(result.lastIndexOf('/') + 1);
+                }
+
+                latestUpload.complete(result);
+                self.set('value', self.get('value').replace('![uploading...]', '![' + filename + ']'));
+            }
+        });
+    },
+
+    didInsertElement: function() {
```

Comment:
There's already a didInsertElement function in this component (line 30). If you move `this.attachFileHandler()` from here to after `this.sendAction('setEditor', this)` in the existing didInsertElement function, it should fix the failing tests.